apps/api/src/app/workflows-v2/util/jsonToSchema.ts

```diff
@@ -67,4 +67,55 @@ function determineSchemaType(value: unknown): JSONSchemaDto {
   }
 }
 
-export { generateJsonSchema, JSONSchemaDto };
+function isMatchingJsonSchema(schema: JSONSchemaDefinition, obj?: Record<string, unknown> | null): boolean {
+  // Ensure the schema is an object with properties
+  if (!obj || !schema || typeof schema !== 'object' || schema.type !== 'object' || !schema.properties) {
+    return false; // If schema is not structured or no properties are defined, assume match
+  }
+
+  // Get the required fields from the schema
+  const requiredFields = schema.required ?? [];
+
+  // Check if all required fields are present in the object
+  const allRequiredFieldsPresent = requiredFields.every((field) => field in obj);
+
+  if (!allRequiredFieldsPresent) return false;
+
+  // Recursively check required fields for nested objects
+  for (const field of requiredFields) {
+    const fieldSchema = schema.properties[field];
+    const fieldValue = obj[field] as Record<string, unknown> | undefined;
+
+    if (typeof fieldSchema === 'object' && fieldSchema.type === 'object' && fieldSchema.properties) {
+      // If a required field is an object, validate its nested structure
+      if (!isMatchingJsonSchema(fieldSchema, fieldValue ?? {})) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function extractMinValuesFromSchema(schema: JSONSchemaDefinition): Record<string, number> {
```

Comment:
The helper function allows extracting the minimum values from the schema, preserving the schema shape.